src/Implementations/AzureTableStorage.cs

```diff
@@ -266,5 +268,64 @@ public async Task TraverseTableEntitiesAsync<T>(string tableName, Action<T> perE
             }
             while (token != null);
         }
+
+        /// <inheritdoc />
+        public async Task DeleteInBatchAsync(
+            string tableName,
+            DateTimeOffset cutOffDate,
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            CloudTable table = await GetTableReferenceAsync(tableName);
+            TableQuery<DynamicTableEntity> query = new TableQuery<DynamicTableEntity>().Where(TableQuery.GenerateFilterConditionForDate("Timestamp", QueryComparisons.LessThan, cutOffDate));
+            IEnumerable<DynamicTableEntity> entities = await GetEntitiesAsync<DynamicTableEntity>(table, query, cancellationToken);
+            await DeleteInBatchAsync<DynamicTableEntity>(table, entities);
+
+        }
+
+        private async Task DeleteInBatchAsync<T>(CloudTable table, IEnumerable<T> tableEntities) where T : ITableEntity, new()
+        {
+            IEnumerable<IGrouping<string, T>> partitionGroups = tableEntities.GroupBy(arg => arg.PartitionKey);
+
+            foreach (IGrouping<string, T> groupedEntities in partitionGroups)
+            {
+                TableBatchOperation tableBatchOperation = new TableBatchOperation();
+                foreach (T item in groupedEntities)
+                {
+                    if (tableBatchOperation.Count() < MAX_BATCH_SIZE)
+                    {
+                        tableBatchOperation.Add(TableOperation.Delete(item));
+                    }
+                    else
+                    {
+                        await table.ExecuteBatchAsync(tableBatchOperation);
+                        tableBatchOperation = new TableBatchOperation
+                         {
+                             TableOperation.Delete(item)
+                         };
+                    }
+                }
+                if (tableBatchOperation.Count() > 0)
+                {
+                    await table.ExecuteBatchAsync(tableBatchOperation);
+                }
+            }
+        }
+
+        private async Task<List<T>> GetEntitiesAsync<T>(
```

Comment:
There could be millions of entries retrieved in these tables...can we not store them all just to delete them? Since we're already iterating over all of the entries, couldn't we just delete them if they're over the age limit instead of storing them? Better yet, could we use filter queries with a Delete operation to delete them without retrieving them? More specifically, is it possible to use `GenerateFilterConditionForDate` with a `TableOperation.Delete` to delete items without retrieving them?

Reply:
No we cannot query for delete operation. The only time we are able to delete without query is if we know RowKey and PartitionKey. But I will do the delete while we get them.